renderdoc/android/jdwp.cpp

```diff
@@ -213,8 +213,29 @@ bool InjectLibraries(const std::string &deviceID, Network::Socket *sock)
 
     if(vulkanLoaderMethod)
     {
-      int32_t slotIdx =
-          conn.GetLocalVariable(vulkanLoaderClass, vulkanLoaderMethod, "librarySearchPath");
+      std::vector<VariableSlot> slots =
+          conn.GetLocalVariables(vulkanLoaderClass, vulkanLoaderMethod);
+
+      int32_t slotIdx = -1, thisSlotIdx = -1;
+      bool slot4Exists = false;
+
+      for(const VariableSlot &s : slots)
+      {
+        if(s.name == "librarySearchPath")
+        {
+          slotIdx = s.slot;
+          break;
+        }
+        else if(s.name == "this")
+          thisSlotIdx = s.slot;
+
+        if(s.slot == 4)
+          slot4Exists = true;
+      }
+
+      // Android 9 changed how slots are indexed, try an offset from this if there is no slot 4
```

Comment:
Minor nit-pick: this comment is wrong because it's not Android 9 and in theory this problem could exhibit on earlier Android versions depending on the particular device. Better to explain that some Android distributions don't have names, and some also have differently numbered slots.

Reply:
It's apparently the JDWP version used since 9 (https://www.pnfsoftware.com/blog/debugging-android-apps-on-android-pie-and-above/)

Reply:
I can believe that it's some bizarre Android internal change I just don't think that it's universal across all of Android 9, before taking patches to use the new layering system on Android 10 the JDWP hooking worked as-is on Android 10 as well.

I don't want the comment to mislead anyone in future. Better to comment directly that the issue is the slot numbering is not necessarily 0-based as previously was assumed.

Reply:
I guess there's really two issues: slot numbering (seems to be consistent on 9) and missing names (doesn't seem as common) and having the names hides the slot numbering change... 

Anyway, changed to "on some newer devices slots are not 0-based, ...".